src/job.rs

```diff
@@ -19,14 +105,59 @@ impl Job {
                     .collect::<Vec<&str>>()
                     .as_slice(),
             )
+            .current_dir(&work_dir)
+            // TODO: this is going to have to retain some environment variables
+            // for software to work correctly. For example, we'll probably need
+            // to provide a fake HOME the way Nix does.
+            .env_clear()
             .output()
+            .context("couldn't run the command")
+    }
+
+    fn tear_down_workspace(&self, work_dir: &Path) -> Result<()> {
```

Comment:
🎨 I expected this to be mainly about deleting things based on the name, until I read what it did; maybe a name like `sync_workspace` might be clearer?